src/Cake.Prca.Issues.MsBuild/MsBuildCodeAnalysisSettings.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Cake.Prca.Issues.MsBuild
+{
+    using System.IO;
+    using Core.IO;
+
+    /// <summary>
+    /// Settings for <see cref="MsBuildCodeAnalysisProvider"/>.
+    /// </summary>
+    public class MsBuildCodeAnalysisSettings
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MsBuildCodeAnalysisSettings"/> class.
+        /// </summary>
+        /// <param name="logFilePath">Path to the the MsBuild log file.
+        /// The log file needs to be in the format as defined by the <paramref name="format"/> parameter.</param>
+        /// <param name="format">Format of the provided MsBuild log file.</param>
+        /// <param name="repositoryRoot">Root path of the repository.</param>
+        public MsBuildCodeAnalysisSettings(FilePath logFilePath, ILogFileFormat format, DirectoryPath repositoryRoot)
```

Comment:
I added two static methods named `MsBuildCodeAnalysisSettingsFromContent` and `MsBuildCodeAnalysisSettingsFromFilePath` to the class and made the ctor protected. I decided against a fully fluent interface though since there is currently no editor supporting intellisense for Cake.